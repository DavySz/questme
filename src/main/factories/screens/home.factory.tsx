import { HomeContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeHomeScreen(): ReactNode {
  return <HomeContainer />;
}
