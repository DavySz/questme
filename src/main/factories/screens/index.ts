export * from "./onboarding-factory";
export * from "./splash-factory";
export * from "./initial-factory";
export * from "./login-factory";
export * from "./sign-up-factory";
