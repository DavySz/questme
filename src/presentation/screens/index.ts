export * from "./home/home.container";
export * from "./splash/splash.container";
export * from "./onboarding/onboarding.container";
export * from "./initial/initial.container";
export * from "./login/login.container";
export * from "./sign-up/sign-up.container";
export * from "./reset-password/reset-password.container";
export * from "./new-password/new-password.container";
export * from "./sign-up-email";
