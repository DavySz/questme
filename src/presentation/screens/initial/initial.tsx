import { Button, ScreenTemplate, Text } from "@presentation/components";
import InitialIllustration from "@presentation/assets/illustrations/initial-illustration.svg";
import LogoIcon from "@presentation/assets/icons/logo-icon.svg";
import {
  Content,
  Header,
  Main,
  Footer,
  TextWrapper,
  ButtonWrapper,
} from "./initial.styles";
import { IInitial } from "./types";

export const Initial = ({
  handleGoToLoginScreen,
  handleGoToSignUpScreen,
}: IInitial): ReturnType<React.FC<IInitial>> => {
  return (
    <ScreenTemplate.Root variant="tertiary">
      <Content testID="initial-screen-content">
        <Header>
          <LogoIcon />
          <Text.Root color="neutral-white" variant="heading-3">
            Queezy
          </Text.Root>
        </Header>
        <Main>
          <InitialIllustration />
        </Main>
        <Footer>
          <TextWrapper>
            <Text.Root color="neutral-black" variant="heading-3">
              Login or Sign Up
            </Text.Root>
            <Text.Root
              align="center"
              color="neutral-gey-2"
              variant="body-normal-regular"
            >
              Login or create an account to take quiz, take part in challenge,
              and more.
            </Text.Root>
          </TextWrapper>
          <ButtonWrapper>
            <Button.Root onPress={handleGoToLoginScreen} testID="button-login">
              <Button.Text>Login</Button.Text>
            </Button.Root>
            <Button.Root
              onPress={handleGoToSignUpScreen}
              testID="button-sign-up"
              variant="secondary"
            >
              <Button.Text>Create an account</Button.Text>
            </Button.Root>
          </ButtonWrapper>
        </Footer>
      </Content>
    </ScreenTemplate.Root>
  );
};
