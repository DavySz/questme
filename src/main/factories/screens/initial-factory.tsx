import { InitialContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeInitialScreen(): ReactNode {
  return <InitialContainer />;
}
