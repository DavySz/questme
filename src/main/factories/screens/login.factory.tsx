import { LoginContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeLoginScreen(): ReactNode {
  return <LoginContainer />;
}
