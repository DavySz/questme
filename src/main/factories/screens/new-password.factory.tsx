import { NewPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeNewPasswordScreen(): ReactNode {
  return <NewPasswordContainer />;
}
