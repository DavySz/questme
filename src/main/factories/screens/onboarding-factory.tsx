import { OnboardingContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeOnboardingScreen(): ReactNode {
  return <OnboardingContainer />;
}
