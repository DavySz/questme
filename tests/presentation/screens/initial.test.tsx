import { InitialContainer } from "@presentation/screens";
import { customRender } from "../../utils/custom-render";
import { fireEvent } from "@testing-library/react-native";
import { Initial } from "@presentation/screens/initial/initial";

describe("Initial", () => {
  test("Should render without errors", () => {
    const tree = customRender(<InitialContainer />);
    expect(tree).toMatchSnapshot();
  });

  test("Should call one time mocked method when button is clicked", () => {
    const handleGoToLoginScreenMock = jest.fn();

    const { getByTestId } = customRender(
      <Initial handleGoToLoginScreen={handleGoToLoginScreenMock} />
    );

    const loginButton = getByTestId("button-login");
    fireEvent.press(loginButton);

    expect(handleGoToLoginScreenMock).toHaveBeenCalledTimes(1);
  });
});
