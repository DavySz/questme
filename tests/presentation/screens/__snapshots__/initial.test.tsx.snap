// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Should render without errors 1`] = `
<View
  style={
    {
      "backgroundColor": "#6A5AE0",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
  testID="bubbles-container"
>
  <View
    left={56}
    style={
      {
        "left": 56,
        "position": "absolute",
        "top": 174,
      }
    }
    top={174}
  >
    <View
      size={48}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
          "height": 48,
          "opacity": 0.1,
          "width": 48,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={200}
    right={78}
    style={
      {
        "bottom": 200,
        "position": "absolute",
        "right": 78,
      }
    }
  >
    <View
      size={20}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 20,
          "opacity": 0.1,
          "width": 20,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={0}
    left={-121}
    style={
      {
        "left": -121,
        "position": "absolute",
      }
    }
  >
    <View
      size={242}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 121,
          "borderBottomRightRadius": 121,
          "borderTopLeftRadius": 121,
          "borderTopRightRadius": 121,
          "height": 242,
          "opacity": 0.1,
          "width": 242,
        }
      }
      type="fill"
    />
  </View>
  <View
    bottom={-83.6}
    left={-209}
    style={
      {
        "bottom": -83.6,
        "left": -209,
        "position": "absolute",
      }
    }
  >
    <View
      size={418}
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 209,
          "borderBottomRightRadius": 209,
          "borderColor": "#C9F2E9",
          "borderStyle": "solid",
          "borderTopLeftRadius": 209,
          "borderTopRightRadius": 209,
          "borderWidth": 1,
          "height": 418,
          "opacity": 0.1,
          "width": 418,
        }
      }
      type="outline"
    />
  </View>
  <View
    right={-121}
    style={
      {
        "position": "absolute",
        "right": -121,
      }
    }
    top={0}
  >
    <View
      size={242}
      style={
        {
          "backgroundColor": "#C9F2E9",
          "borderBottomLeftRadius": 121,
          "borderBottomRightRadius": 121,
          "borderTopLeftRadius": 121,
          "borderTopRightRadius": 121,
          "height": 242,
          "opacity": 0.1,
          "width": 242,
        }
      }
      type="fill"
    />
  </View>
  <View
    right={-209}
    style={
      {
        "position": "absolute",
        "right": -209,
        "top": -83.6,
      }
    }
    top={-83.6}
  >
    <View
      size={418}
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 209,
          "borderBottomRightRadius": 209,
          "borderColor": "#C9F2E9",
          "borderStyle": "solid",
          "borderTopLeftRadius": 209,
          "borderTopRightRadius": 209,
          "borderWidth": 1,
          "height": 418,
          "opacity": 0.1,
          "width": 418,
        }
      }
      type="outline"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "marginTop": 54,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
    variant="tertiary"
  >
    <View
      style={
        {
          "gap": 16,
        }
      }
    >
      <SvgMock />
      <Text
        color="neutral-white"
        style={
          {
            "color": "#FFF",
            "fontFamily": "Rubik_500Medium",
            "fontSize": 24,
            "lineHeight": 36,
          }
        }
        variant="heading-3"
      >
        Queezy
      </Text>
    </View>
    <View
      style={{}}
    >
      <SvgMock />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "gap": 24,
          "justifyContent": "center",
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "gap": 8,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          color="neutral-black"
          style={
            {
              "color": "#0C092A",
              "fontFamily": "Rubik_500Medium",
              "fontSize": 24,
              "lineHeight": 36,
            }
          }
          variant="heading-3"
        >
          Login or Sign Up
        </Text>
        <Text
          align="center"
          color="neutral-gey-2"
          style={
            {
              "color": "#858494",
              "fontFamily": "Rubik_400Regular",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "center",
            }
          }
          variant="body-normal-regular"
        >
          Login or create an account to take quiz, take part in challenge, and more.
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "gap": 16,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#6A5AE0",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
          testID="button-login"
        >
          <Text
            color="neutral-white"
            style={
              {
                "color": "#FFF",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-medium"
          >
            Login
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#EFEEFC",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <Text
            color="primary"
            style={
              {
                "color": "#6A5AE0",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-medium"
          >
            Create an account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
